src/Microsoft.Health.Fhir.Azure/AnonymizationConfigurationArtifactProvider.cs

```diff
@@ -165,7 +165,7 @@ private async Task FetchConfigurationFromBlob(string configName, string eTag, St
             BlobClient blob = container.GetBlobClient(configName);
             if (await blob.ExistsAsync(cancellationToken))
             {
-                if (CheckConfigurationIsTooLarge(blob.GetProperties(cancellationToken: cancellationToken).Value.ContentLength))
+                if (CheckConfigurationIsTooLarge(size: blob.GetPropertiesAsync(cancellationToken: cancellationToken).GetAwaiter().GetResult().Value.ContentLength))
```

Comment:
Is there a reason why you chose to use GetAwaiter instead of using await? If it is an issue with using it in the method call, then I would just create a temp variable above that if statement and make the async call there. If you stick with the GetAwaiter then there is the potential for deadlocks as it isn't asynchronous.